packages/ui-components/src/components/Table/TableRow.js

```diff
@@ -70,7 +70,7 @@ export const StyledTableRow = styled(MuiTableRow)`
 `;
 
 export const TableRow = React.memo(({ columns, rowData, className, onRowClick }) => (
-  <StyledTableRow className={className} onClick={onRowClick}>
+  <StyledTableRow className={className} onClick={e => onRowClick(e, rowData)}>
```

Comment:
Hi @tcaiger , I'd like to check this change through you... Although I could find other ways to manage state related to a clicked row (eg using a Provider), I believe that the above behaviour is reasonable, and perhaps expected. Keen to hear your thoughts 🙂

Also, this change _should_ be safe since `onClick` is natively called with just one argument. So unless if a client does something unexpected, the new `rowData` parameter shouldn't shadow any argument in existing usages of `onRowClick`

Reply:
Speaking of safe changes - unit tests saved my bacon! https://github.com/beyondessential/tupaia/pull/2545/commits/d46ae910d6dd771889e0323fd44d4365f85b863a

Reply:
Yes I think this is a great idea!

Reply:
Eventually I will use `Context` (for other reasons). So I'll revert this change in another PR, but let's keep it in mind in case it would be useful in the future!